### DISCUSSION TITLE
Variant on the define_word script

### DISCUSSION THREAD

**Discussion by PietJanHein** [BreadOnPenguins/scripts#3] [Show and tell]
Not always it is possible to select a word by blocking it with the mouse, for example if a person is gaming or hears or sees (subtitles) a word in a movie/TV-show or song. So I changed the define_word script so that it also works from the terminal. The changes are quite trivial, I will paste the changed lines here and show both the script and how to use it in a screenshot. Feel free to add it to your repository and name it however you want. 

Changed lines: 
word=$1 #$1 refers to the first argument given when you start the script

bold=$(tput bold) #This makes it possible to print text bold with echo, required for visual clarity
normal=$(tput sgr0) #This makes it possible to reset the text to normal again with echo
echo "${bold}Definition of $word"
echo "${normal}$def"

![image](https://github.com/user-attachments/assets/923c4c89-2423-4f7a-aafd-c8fdff0f9e0f)

![image](https://github.com/user-attachments/assets/a12cd9d4-f41f-4a06-a7c3-a3968fbe86e2)

**Comment by PietJanHein** [comment]
I have 1 more variant, this time to ask the user to input the word instead of giving it as an input argument. Why do that? It saves the user from having to type the directory every time, he can simply make a global hotkey and use the -e option to open up the terminal and run that script. 

This is the change. 
read -p "Enter the word: " word

If you use the Alacritty terminal and the script is in ~/scripts and named word3 then simply set the global hotkey to this: alacritty -e ~/scripts/word3

Finally to prevent the terminal from closing down immediately after showing the output you have to add these lines on the bottom: 
read var
        if [[ -n $var ]]
        then
                continue
        fi
done

This keeps the terminal in a loop until the enter-key is hit so that the terminal only closes after that and not directly before the user can read the output..

**Comment by BreadOnPenguins** [comment]
Thanks so much for the suggestions and ideas! Just added the ability for it to work from terminal, and commented out the option for echo in case it is wanted.

Last idea is a great modification but since it would alter the functionality significantly I'll leave the script as is, but feel free to fork and add :-) 

Thanks again!
